src/test/java/com/acme/conferencesystem/voting/persistence/VoteRepositoryTest.java

```diff
@@ -0,0 +1,63 @@
+package com.acme.conferencesystem.voting.persistence;
+
+import com.acme.conferencesystem.ContainerConfig;
+import com.acme.conferencesystem.cfp_proposals.persistence.ProposalEntity;
+import com.acme.conferencesystem.cfp_proposals.persistence.ProposalRepository;
+import com.acme.conferencesystem.users.persistence.UserEntity;
+import com.acme.conferencesystem.users.persistence.UsersRepository;
+import org.assertj.core.api.Assertions;
+import org.instancio.Instancio;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Import;
+import org.springframework.modulith.test.ApplicationModuleTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.instancio.Select.field;
+
+@ApplicationModuleTest(mode = ApplicationModuleTest.BootstrapMode.DIRECT_DEPENDENCIES)
+@Import(ContainerConfig.class)
+class VoteRepositoryTest {
+
+    @Autowired
+    UsersRepository usersRepository;
+
+    @Autowired
+    ProposalRepository proposalRepository;
+
+    @Autowired
+    VoteRepository voteRepository;
+    private ProposalEntity persistedProposal;
+    private UserEntity persistedUser;
+
+    @BeforeEach
+    void setUp() {
+
+        persistedUser = usersRepository.save(
+                Instancio.of(UserEntity.class)
+                        .ignore(field(UserEntity::id))
+                        .create());
+
+        persistedProposal = proposalRepository.save(
+                Instancio.of(ProposalEntity.class)
+                        .ignore(field(ProposalEntity::id))
+                        .set(field(ProposalEntity::speakerId), persistedUser.id())
+                        .create());
```

Comment:
Not very happy using the repositories of the other modules in the voting repository test. But well we can discuss about it in the future ;)

Reply:
I'm thinking about the future migration into a microservice